clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -3574,6 +3574,16 @@ def DeleteArrayOp : CIR_Op<"delete.array">,
 // CallOp and TryCallOp
 //===----------------------------------------------------------------------===//
 
+def SE_All : I32EnumAttrCase<"All", 1, "all">;
+def SE_Pure : I32EnumAttrCase<"Pure", 2, "pure">;
+def SE_Const : I32EnumAttrCase<"Const", 3, "const">;
+
+def SideEffect : I32EnumAttr<
+    "SideEffect", "allowed side effects of a function",
+    [SE_All, SE_Pure, SE_Const]> {
+  let cppNamespace = "::cir";
```

Comment:
Please add a bit more doc on what are the source level things that map to these!

Reply:
Updated.

Reply:
Awesome, minor nit - probably not necessary to mention the enum names (and maybe confusing?), so minor tweak: so "`all`: The callee"  and similar for others is enough. A quick usage example is also probably good: `%3 = cir.call @add(%0, %1) : (!s32i, !s32i) -> !s32i side_effect(pure)`

Reply:
Updated.